src/interchange/avro.rs

```diff
@@ -929,12 +929,15 @@ impl Encoder {
                     ScalarType::TimestampTz => {
                         Value::Timestamp(datum.unwrap_timestamptz().naive_utc())
                     }
+                    // Duration Avro format: https://avro.apache.org/docs/current/spec.html#Duration
                     ScalarType::Interval => Value::Fixed(12, {
                         let iv = datum.unwrap_interval();
                         let mut buf = Vec::with_capacity(12);
+                        let days = iv.days() as i128;
+                        let sub_day_ns = iv.duration % (days * 24 * 60 * 60 * 1_000_000_000);
                         buf.extend(&(iv.months as i32).to_le_bytes());
-                        buf.extend(&0i32.to_le_bytes());
-                        buf.extend(&(iv.duration.as_millis() as i32).to_le_bytes());
+                        buf.extend(&(days as i32).to_le_bytes());
+                        buf.extend(&((sub_day_ns / 1_000_000) as i32).to_le_bytes());
```

Comment:
it seems possibly (locally, at least) that this could exceed an `i32` if `days` is large. I'm not sure what `days` is, but some notes on why this is all ok here would help me out!

Reply:
Annoyingly, this looks like it's going to be bad for Avro for the time being because it actually only supports `u32`--glad you made this note. `i32` suffices because it matches the maximum number of days Postgres allows for intervals, though, which is good. At any rate, I left a note that will need to be cleaned up a bit once we determine what to do here.